runtime/js/40_testing.js

```diff
@@ -240,6 +241,24 @@ finishing test case.`;
       only: only.length > 0,
     });
 
+    if (shuffle) {
+      // http://en.wikipedia.org/wiki/Linear_congruential_generator
+      const nextInt = (function (state) {
+        const m = 0x80000000;
+        const a = 1103515245;
+        const c = 12345;
+
+        return function (max) {
+          return state = ((a * state + c) % m) % max;
+        };
+      }(shuffle));
+
+      for (let i = pending.length - 1; i > 0; i--) {
+        const j = 1 + nextInt(pending.length - 1);
```

Comment:
If you're trying to implement [Fisher-Yates](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle), this line should probably be:
```suggestion
        const j = nextInt(i);
```

Reply:
I think that would cause it to prefer shuffling to items with a lower index? (So the `0` index value would always stay the same, I think)

That said, why is it `1 +`? Am I reading it correctly that `j` would never be `0`? I'm also getting `.5` values (ex. `1.5`, `3.5`, etc) trying out this code... `nextInt` doesn't seem to be returning an integer.

Reply:
> I think that would cause it to prefer shuffling to items with a lower index?

Yes, it operates on a shrinking view of the list.

> So the 0 index value would always stay the same, I think

No, unless `nextInt` never returns zero.